junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/TempDirectory.java

```diff
@@ -136,7 +138,8 @@ public boolean supportsParameter(ParameterContext parameterContext, ExtensionCon
 	public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext) {
 		Class<?> parameterType = parameterContext.getParameter().getType();
 		assertSupportedType("parameter", parameterType);
-		return getPathOrFile(parameterType, extensionContext);
+		String id = parameterContext.findAnnotation(TempDir.class).map(TempDir::value).orElse("");
```

Comment:
❓ Should we validate whether `TempDir.value()` returns `null`?

Reply:
Values returned by an annotation element are never `null`: https://stackoverflow.com/a/41309428/1431016